internal/podman/podman.go

```diff
@@ -58,11 +64,87 @@ func runPodman(ctx context.Context, args ...string) (bytes.Buffer, error) {
 	cmd.Stdout = &stdout
 	cmd.Stderr = &stderr
 	if err := cmd.Run(); err != nil {
+		var exiterr *exec.ExitError
+		if errors.As(err, &exiterr); exiterr.ExitCode() == 8 {
+			return stdout, fmt.Errorf(stderr.String())
+		}
 		return stdout, fmt.Errorf("podman error (args=%v) (stderr=%v) (error=%w)", args, stderr.String(), err)
 	}
 	return stdout, nil
 }
 
+func ScanJava(ctx context.Context, image string, javaDisabledAlgorithms []string) (string, error) {
+	javaFilePath, _, err := releases.GetJavaFile()
+	defer os.Remove(javaFilePath)
+	if err != nil {
+		return "", err
+	}
+
+	algFilePath, algFile, err := releases.GetAlgorithmFile(javaDisabledAlgorithms)
+	defer os.Remove(algFilePath)
+	if err != nil {
+		return "", err
+	}
+
+	data, err := Inspect(ctx, image, "--format", "{{index  .Config.Entrypoint}}|{{index  .Config.Cmd}}|{{index  .Config.WorkingDir}}")
+	if err != nil {
+		return "", err
+	}
+	parts := strings.Split(data, "|")
+
+	entry := strings.TrimPrefix(strings.TrimSpace(parts[0]), "[")
+	entry = strings.TrimSuffix(entry, "]")
+	cmd := strings.TrimPrefix(strings.TrimSpace(parts[1]), "[")
+	cmd = strings.TrimSuffix(cmd, "]")
+
+	jInfo := &types.JavaComponent{}
+	jInfo.Entrypoint = strings.Split(entry, ",")
+	jInfo.Cmd = strings.Split(cmd, ",")
+	jInfo.WorkingDir = strings.TrimSpace(parts[2])
+
+	// this was done because java versions before 1.11 cannot execute .Java src files w/o compiling first.
```

Comment:
Please start a comment with a Capital letter, so it reads like a whole sentence.